token-generator/src/main/java/uk/gov/hmcts/auth/totp/StrangeTotpAuthenticator.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.auth.totp;
+
+import com.warrenstrange.googleauth.GoogleAuthenticator;
+import com.warrenstrange.googleauth.IGoogleAuthenticator;
+
+public class StrangeTotpAuthenticator implements TotpAuthenticator {
```

Comment:
Perhaps `TotpAuthenticatorImpl`?

Reply:
This class uses `GoogleAuthenticator` implementation from Warren Strange, and I couldn't resist naming it after him.

Reply:
Ok, following team/org naming profile. Still, previous name was **WAY** better.